crates/indexer/src/models/v2_objects.rs

```diff
@@ -117,38 +105,55 @@ impl Object {
         }
     }
 
-    /// This should never really happen since it's very difficult to delete the entire resource group
-    /// currently. We actually need a better way of detecting whether an object is deleted since there
-    /// is likely no delete resource write set change.
-    fn from_delete_resource(
+    /// This handles the case where the entire object is deleted
+    /// TODO: We need to detect if an object is only partially deleted
+    /// using KV store
+    pub fn from_delete_resource(
         delete_resource: &DeleteResource,
         txn_version: i64,
         write_set_change_index: i64,
+        object_mapping: &HashMap<CurrentObjectPK, CurrentObject>,
+        conn: &mut PgPoolConnection,
     ) -> anyhow::Result<Option<(Self, CurrentObject)>> {
-        if delete_resource.resource.to_string() == "0x1::object::ObjectCore" {
+        if delete_resource.resource.to_string() == "0x1::object::ObjectGroup" {
```

Comment:
What's object group and why do we need to change this?

Reply:
the old thing wasn't working hahaha. I assuemd it would be ObjectCore but actually when it's deleted it's ObjectGroup. It was just buggy before.

Reply:
that's weird that the resource type is different for writes and deletes.